packages/shared/lib/core/wallet/utils/outputs/getAliasActivityData.ts

```diff
@@ -0,0 +1,58 @@
+import { COIN_TYPE } from '@core/network'
+import { activeProfile } from '@core/profile'
+import { get } from 'svelte/store'
+import { IProcessedTransaction, IAliasActivityData } from '../../interfaces'
+import {
+    getAmountFromOutput,
+    getGovernorAddressFromOutput,
+    getNativeTokenFromOutput,
+    getStorageDepositFromOutput,
+    getStateControllerAddressFromOutput,
+    convertHexAddressToBech32,
+} from '..'
+import { ActivityType, AliasType } from '@core/wallet/enums'
+import { ADDRESS_TYPE_ALIAS, NEW_ALIAS_ID, OUTPUT_TYPE_ALIAS } from '@core/wallet/constants'
+import { OutputTypes } from '@iota/types'
+import { Converter } from '@lib/converter'
+import { Blake2b } from '@iota/crypto.js'
+
+export function getAliasActivityData(processedTransaction: IProcessedTransaction): IAliasActivityData {
+    const { outputs } = processedTransaction
+    const { output, outputId } = outputs.find((output) => output.output.type === OUTPUT_TYPE_ALIAS)
+
+    const nativeToken = getNativeTokenFromOutput(output)
+    const assetId = nativeToken?.id ?? String(COIN_TYPE[get(activeProfile).networkProtocol])
+
+    const storageDeposit = getAmountFromOutput(output) + getStorageDepositFromOutput(output).storageDeposit
+    const governorAddress = getGovernorAddressFromOutput(output)
+    const stateControllerAddress = getStateControllerAddressFromOutput(output)
+    const aliasId = getAliasId(output, outputId)
+    const aliasType =
+        output.type === OUTPUT_TYPE_ALIAS && output.aliasId === NEW_ALIAS_ID ? AliasType.Created : AliasType.Other
+
+    return {
+        type: ActivityType.Alias,
+        outputId,
+        assetId,
+        aliasId,
+        aliasType,
+        storageDeposit,
+        governorAddress,
+        stateControllerAddress,
+    }
+}
+
+function getAliasId(output: OutputTypes, outputId: string): string {
```

Comment:
Might we need this function elsewhere? So it can be in its own file/exported?

Reply:
no its only used for the initial conversion from output -> activity